src/main/scala/org/allenai/pipeline/Producer.scala

```diff
@@ -75,20 +75,26 @@ trait Producer[+T] extends PipelineStep with CachingEnabled with Logging {
   }
 
   private var initialized = false
-  protected[this] var timing: Option[Duration] = None
-  protected[this] var executionMode: Duration => ExecutionInfo = Executed
+  protected[this] var timing: Duration = Duration.Undefined
+  private[this] var _executionMode: Duration => ExecutionInfo = NotRequested
+  protected[this] def executionMode: Duration => ExecutionInfo = _executionMode
+  protected[this] def executionMode_=(newValue: Duration => ExecutionInfo) = {
+    if (_executionMode == NotRequested) {
```

Comment:
Don't change the value of `executionMode` unless the value is currently `NotRequested`